x-pack/legacy/plugins/maps/public/layers/styles/vector/components/legend/vector_icon.js

```diff
@@ -48,16 +46,16 @@ export class VectorIcon extends Component {
 
     if (this.state.isLinesOnly) {
       const style = {
-        stroke: extractColorFromStyleProperty(this.props.lineColor, 'grey'),
+        stroke: this.props.getColorForProperty(VECTOR_STYLES.LINE_COLOR, true),
```

Comment:
This is really weird as a callback. Why not just pass in fill and line color as properties? If its because getting isLines only is async then lets fix that before this PR so these components can be more straight forward.